### PR TITLE
Improve README guidance for release process

### PR DESCRIPTION
**Pull request by aduth**
**Why**:

- Changelog should be edited to replace "Unreleased" with the release version.
- Main branch is protected, so you cannot commit the version bump directly. Improve text to include steps for creating a branch, opening a pull request, and merging before creating a new release.
- Dry-run testing is all well and good, but there wasn't any guidance around what exactly one should be looking for in considering whether "everything looks alright". Add a couple tips on what to look for.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,15 +60,21 @@ When you're ready to release a new version of the `identity-style-guide` package
...
+5. Change the "Unreleased" heading in `CHANGELOG.md` to the version you decided in Step 3. Commit this change to your new branch.
+6. Run `npm version` to bump the package version, passing one of `patch`, `minor`, or `major` depending on what you had decided in Step 3 for the next version.
+   - Example: `npm version patch`
+   - A new version will be created. This will update `package.json` and `package-lock.json` automatically and create a commit.
+7. Do a trial publish by running `npm publish --dry-run`.
    - No need to run any special build steps — the publish script will lint the source JavaScript and Sass files, and clean and re-build all assets before including them in the published package.
-6. If everything looks alright, continue with publishing by running `npm publish`.
-7. Create a new release on the [GitHub "Releases" page](https://github.com/18F/identity-style-guide/releases).
+   - Consider: In the files listed, are there any that should or shouldn't be included? Does the version match what you expect?
+8. If everything looks alright, continue with publishing by running `npm publish`.
```
Comment (zachmargolis) [inline README.md:75]: We have a handbook page that links to a google doc that lists who has access to NPM to be able to publish, should we link to that from here?

https://handbook.login.gov/articles/accounts.html

**Review by zachmargolis — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (aduth) [reply README.md]: > We have a handbook page that links to a google doc that lists who has access to NPM to be able to publish, should we link to that from here?

I shuffled this a bit in 39fc4e0 to include a leading "prerequisites" blurb of text:

>Before starting, make sure that all changes intended for release should be merged into the `main` branch. You will need permissions to publish the package to npm. Check current package owners by running `npm owner ls` or by consulting the list of admins through the [Services and Accounts handbook page](https://handbook.login.gov/articles/accounts.html). If you do not have access, contact an owner to have access granted or to publish on your behalf.

**Review by solipet — APPROVED** [review]
Nice.

**Pull request merged**